src/object.cr

```diff
@@ -488,20 +488,31 @@ class Object
       \{% for name in names %}
         \{% if name.is_a?(TypeDeclaration) %}
           {{var_prefix}}\{{name}}?
-          \{% name = name.var %}
-        \{% end %}
 
-        def {{method_prefix}}\{{name.id}}?
-          {{var_prefix}}\{{name.id}}
-        end
+          def {{method_prefix}}\{{name.var.id}}? : \{{name.type}}?
```

Comment:
To prevent some duplication you could prob do like `{% if name.is_a?(TypeDeclaration) %} : \{{name.type}}? {% end %}`  Or something along those lines.

Reply:
I had that originally, but then I noticed that all the other macros use this pattern, so I made it consistent. I'd be fine to change it.